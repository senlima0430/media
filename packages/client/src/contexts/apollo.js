import ApolloClient from 'apollo-boost'

export const client = new ApolloClient({
  uri: `http://${window.location.hostname}:8080/graphql`,
})
