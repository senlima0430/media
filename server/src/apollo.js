import { ApolloServer } from 'apollo-server-express'
import { importSchema } from 'graphql-import'
import { makeExecutableSchema } from 'graphql-tools'
import models from './models'

import { resolvers } from './resolvers'

const typeDefs = importSchema(`${__dirname}/typeDefs/schema.graphql`)

const schema = makeExecutableSchema({
  typeDefs,
  resolvers,
})

export const apollo = new ApolloServer({
  schema,
  playground: process.env.NODE_ENV === 'development',
  context: async ({ req, connection }) => {
    if (connection) {
      return connection.context
    } else {
      return { models }
    }
  },
})
