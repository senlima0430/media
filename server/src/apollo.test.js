import gql from 'graphql-tag'
import { createTestClient } from 'apollo-server-testing'
import { apollo } from './apollo'

describe('Queries', () => {
  it('should response exist', async () => {
    const { query } = createTestClient(apollo)
    const res = await query({
      query: gql`
        query probe {
          probe
        }
      `,
    })

    expect(res).toMatchSnapshot()
  })
})
