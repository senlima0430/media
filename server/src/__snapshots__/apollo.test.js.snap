// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries should response exist 1`] = `
Object {
  "data": Object {
    "probe": "exist",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
